test/integration/controller/core/clusterqueue_controller_test.go

```diff
@@ -71,120 +59,212 @@ var _ = ginkgo.Describe("ClusterQueue controller", func() {
+				var newCQ kueue.ClusterQueue
+				gomega.Expect(k8sClient.Get(ctx, client.ObjectKeyFromObject(cq), &newCQ)).To(gomega.Succeed())
+				return newCQ.GetFinalizers()
+			}, framework.Timeout, framework.Interval).Should(gomega.Equal([]string{core.ClusterQueueFinalizerName}))
+		})
+	})
+
+	ginkgo.When("Deleting clusterQueues", func() {
+		var (
+			cq    *kueue.ClusterQueue
+			queue *kueue.Queue
+		)
+
+		ginkgo.BeforeEach(func() {
+			cq = testing.MakeClusterQueue("foo-cq").Obj()
+			queue = testing.MakeQueue("queue", ns.Name).ClusterQueue(cq.Name).Obj()
+			gomega.Expect(k8sClient.Create(ctx, queue)).To(gomega.Succeed())
+			gomega.Expect(k8sClient.Create(ctx, cq)).To(gomega.Succeed())
+		})
+
+		ginkgo.It("Should delete clusterQueues successfully when no admitted workloads are running", func() {
+			gomega.Expect(framework.DeleteClusterQueue(ctx, k8sClient, cq)).To(gomega.Succeed())
+			gomega.Eventually(func() bool {
+				var newCQ kueue.ClusterQueue
+				err := k8sClient.Get(ctx, client.ObjectKeyFromObject(cq), &newCQ)
+				return err != nil && apierrors.IsNotFound(err)
+			}, framework.Timeout, framework.Interval).Should(gomega.BeTrue())
+		})
+
+		ginkgo.It("Should stuck in termination until admitted workloads finished running", func() {
+			ginkgo.By("Admit workloads")
+			admission := testing.MakeAdmission(cq.Name).Obj()
+			wl := testing.MakeWorkload("workload", ns.Name).Queue(queue.Name).Admit(admission).Obj()
+			gomega.Expect(k8sClient.Create(ctx, wl)).To(gomega.Succeed())
+
+			ginkgo.By("Delete clusterQueue")
+			gomega.Expect(framework.DeleteClusterQueue(ctx, k8sClient, cq)).To(gomega.Succeed())
+			var newCQ kueue.ClusterQueue
+			gomega.Eventually(func() []string {
+				gomega.Expect(k8sClient.Get(ctx, client.ObjectKeyFromObject(cq), &newCQ)).To(gomega.Succeed())
+				return newCQ.GetFinalizers()
+			}, framework.Timeout, framework.Interval).Should(gomega.Equal([]string{core.ClusterQueueFinalizerName}))
+
+			ginkgo.By("Finish workloads")
 			gomega.Eventually(func() error {
 				var newWL kueue.Workload
-				gomega.Expect(k8sClient.Get(ctx, client.ObjectKeyFromObject(w), &newWL)).To(gomega.Succeed())
+				gomega.Expect(k8sClient.Get(ctx, client.ObjectKeyFromObject(wl), &newWL)).To(gomega.Succeed())
 				newWL.Status.Conditions = append(newWL.Status.Conditions, kueue.WorkloadCondition{
 					Type:   kueue.WorkloadFinished,
 					Status: corev1.ConditionTrue,
 				})
 				return k8sClient.Status().Update(ctx, &newWL)
-			}, framework.Timeout, framework.Interval).Should(gomega.Succeed())
-		}
-		gomega.Eventually(func() kueue.ClusterQueueStatus {
-			var updatedCq kueue.ClusterQueue
-			gomega.Expect(k8sClient.Get(ctx, client.ObjectKeyFromObject(clusterQueue), &updatedCq)).To(gomega.Succeed())
-			return updatedCq.Status
-		}, framework.Timeout, framework.Interval).Should(testing.Equal(kueue.ClusterQueueStatus{
-			UsedResources: emptyUsedResources,
-		}))
+			}, framework.Timeout, framework.Interval).Should(gomega.BeNil())
```

Comment:
You should create another workload at this point and check that is not admitted.

That's probably flaky with this version of the PR.